src/Processors/Transforms/FillingTransform.cpp

```diff
@@ -184,18 +196,20 @@ FillingTransform::FillingTransform(
         fill_column_positions.push_back(block_position);
 
         auto & descr = filling_row.getFillDescription(i);
-        const auto & type = header_.getByPosition(block_position).type;
+
+        const Block & output_header = getOutputPort().getHeader();
```

Comment:
Taking it from the output header is unneeded, as it is currently identical to the input header by the data types, but I was not sure about it and added this code. It is more natural, so worth keeping this change.